File: CategoryA/1792A.c
// Daniel.Soto - CosmologyAndGravitation_GravitationalWaves_LVK
// Contact: daniel.soto1@udea.edu.co

#include <stdio.h>

int main(void) {
    int t;
    scanf("%d", &t);
    
    for (int i = 0; i < t; i++) {
        int n;
        scanf("%d", &n);
        
        int h[n];
        for (int j = 0; j < n; j++) {
            scanf("%d", &h[j]);
        }
        
        int contador = 0;
        for (int j = 0; j < n; j++) {
            if (h[j] == 1) { contador++;
            }
        }
        
        printf("%d\n", n - contador / 2);
    }
    
    return 0;
}

File: CategoryA/1792A.py
# Daniel.Soto - CosmologyAndGravitation_GravitationalWaves_LVK
# Contact: daniel.soto1@udea.edu.co

t = int(input())

for _ in range(t):
	n = int(input())	
	h = list(map(int, input().split()))
	
	contador = 0

	for i in h:
		if i==1:
			contador +=1

	print(n - contador//2)
